### PR TITLE
Feat/use parameter name as pk

### PR DESCRIPTION
**Pull request by tonsV2**

### CODE REVIEW

File: pkg/instance/handler.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func (h Handler) Create(c *gin.Context) {
 }
 
 type ParameterRequest struct {
-	StackParameterID uint   `json:"stackParameterId" binding:"required"`
+	StackParameterID string `json:"stackParameterId" binding:"required"`
```
Comment (teleivo) [inline pkg/instance/handler.go:107]: Can/should we rename this to `StackParameter` (at some point)?

File: pkg/instance/handler.go [hunk 2]
```diff
@@ -192,29 +192,31 @@ func (h Handler) Deploy(c *gin.Context) {
 	c.JSON(http.StatusCreated, instance)
 }
 
-func convertRequiredParameters(instanceID uint, requestParameters []ParameterRequest) []model.InstanceRequiredParameter {
+func convertRequiredParameters(instance *model.Instance, requestParameters []ParameterRequest) []model.InstanceRequiredParameter {
 	if len(requestParameters) > 0 {
 		var parameters = make([]model.InstanceRequiredParameter, len(requestParameters))
 		for i, parameter := range requestParameters {
+			stackRequiredParameter := model.StackRequiredParameter{ID: parameter.StackParameterID}
```
Comment (teleivo) [inline pkg/instance/handler.go:199]: I would inline this (and in function below) into the struct literal if not used anywhere else. saves us naming/reading/maintaining a var in our head 😋

File: pkg/instance/handler.go [hunk 3]
```diff
@@ -192,29 +192,31 @@ func (h Handler) Deploy(c *gin.Context) {
 	c.JSON(http.StatusCreated, instance)
 }
 
-func convertRequiredParameters(instanceID uint, requestParameters []ParameterRequest) []model.InstanceRequiredParameter {
+func convertRequiredParameters(instance *model.Instance, requestParameters []ParameterRequest) []model.InstanceRequiredParameter {
```
Comment (teleivo) [inline pkg/instance/handler.go:195]: Why did you change the parameter from uint to *model.Instance? All you need in here is the uint, right?

File: pkg/instance/helmfile.go [hunk 4]
```diff
@@ -95,12 +95,12 @@ func (h helmfileService) configureInstanceEnvironment(accessToken string, instan
...
 
 	for _, parameter := range instance.RequiredParameters {
-		instanceEnv := fmt.Sprintf("%s=%s", parameter.StackRequiredParameter.Name, parameter.Value)
+		instanceEnv := fmt.Sprintf("%s=%s", parameter.StackRequiredParameter.ID, parameter.Value)
 		cmd.Env = append(cmd.Env, instanceEnv)
 	}
 
 	for _, parameter := range instance.OptionalParameters {
-		instanceEnv := fmt.Sprintf("%s=%s", parameter.StackOptionalParameter.Name, parameter.Value)
+		instanceEnv := fmt.Sprintf("%s=%s", parameter.StackOptionalParameter.ID, parameter.Value)
```
Comment (teleivo) [inline pkg/instance/helmfile.go:103]: why rename it from Name to ID?

File: pkg/stack/loader.go [hunk 5]
```diff
@@ -77,7 +77,7 @@ func LoadStacks(stackService Service) {
 }
 
 func extractRequiredParameters(file []byte) []string {
-	regexStr := "\\{\\{[ ]?requiredEnv[ ]?\"(.*?)\"[ ]?\\}\\}"
+	regexStr := "\\{\\{[ ]?requiredEnv[ ]?\"(.*?)\".*\\}\\}"
```
Comment (teleivo) [inline pkg/stack/loader.go:80]: does that relate to switching from ids to names?

File: scripts/deploy-job.sh [hunk 6]
```diff
@@ -15,26 +15,7 @@ PAYLOAD='{\"databaseId\": \"11\"}'
 echo "{
   \"requiredParameters\": [
     {
-      \"stackParameterId\": 5,
-      \"value\": \"$JOB\"
-    }
-  ],
-  \"optionalParameters\": [
-    {
-      \"stackParameterId\": 10,
```
Comment (teleivo) [inline scripts/deploy-job.sh:24]: why did you delete the optional params?

File: scripts/dhis2-core-deploy.sh [hunk 7]
```diff
@@ -26,21 +19,21 @@ echo "{
   \"stackId\": 1,
   \"optionalParameters\": [
     {
-      \"stackParameterId\": $READINESS_PROBE_INITIAL_DELAY_SECONDS_PARAMETER_ID,
+      \"stackParameterId\": \"READINESS_PROBE_INITIAL_DELAY_SECONDS\",
```
Comment (teleivo) [inline scripts/dhis2-core-deploy.sh:22]: 🎉 nice

File: scripts/dhis2-deploy.sh [hunk 8]
```diff
@@ -2,23 +2,13 @@
...
-
-IMAGE_TAG="2.36.0-tomcat-8.5.34-jre8-alpine"
-IMAGE_TAG_PARAMETER_ID=1
-
-DATABASE_ID=1
-DATABASE_ID_PARAMETER_ID=1
+IMAGE_TAG=2.36.0-tomcat-8.5.34-jre8-alpine
+DATABASE_SIZE=30Gi
+PGADMIN_INSTALL=false
+DATABASE_ID=3
```
Comment (teleivo) [inline scripts/dhis2-deploy.sh:11]: DATABASE_ID changed from 1 to 3, why?

File: stacks/dhis2-core/helmfile.yaml [hunk 9]
```diff
@@ -27,7 +27,7 @@ releases:
       - dhisConfig: |
           connection.dialect = org.hibernate.dialect.PostgreSQLDialect
           connection.driver_class = org.postgresql.Driver
-          connection.url = jdbc:postgresql://{{ requiredEnv "DATABASE_HOSTNAME" }}/{{ env "DATABASE_DATABASE" | default "dhis2" }}
+          connection.url = jdbc:postgresql://{{ requiredEnv "INSTANCE_NAME" }}-database-postgresql/{{ env "DATABASE_DATABASE" | default "dhis2" }}
```
Comment (teleivo) [inline stacks/dhis2-core/helmfile.yaml:30]: should this be changed? if so in the dhis2-core-deploy script DATABASE_HOSTNAME is still provided as required parameter

**Review by teleivo — COMMENTED** [review]
I've added some questions. I'll need a bit more time to look at the pkg/model/stack.go GORM stuff. That is a bit hard to decipher 😂

File: pkg/instance/handler.go (hunk 2 above)
Comment (tonsV2) [reply pkg/instance/handler.go:199]: Done!

File: scripts/deploy-job.sh (hunk 6 above)
Comment (tonsV2) [reply scripts/deploy-job.sh:24]: I actually didn't. The script, by mistake, had two echo commands and only the second piped its output to the http command.

It should have been addressed in a separate commit but if you don't object I'll include it here rather than reverting it.

File: scripts/dhis2-core-deploy.sh (hunk 7 above)
Comment (tonsV2) [reply scripts/dhis2-core-deploy.sh:22]: Damn right!

File: pkg/stack/loader.go (hunk 5 above)
Comment (teleivo) [reply pkg/stack/loader.go:80]: then what is it 😋 its easier to review a PR if it only contains whats in the PR title/description/task.

**Review by teleivo — APPROVED** [review]
can't wait to use the scripts in my environment 🎉 🥇
